### PR TITLE
Fix inconsistency between parallel and serial coset ffts

### PR DESCRIPTION
**Pull request by ValarDragon**
## Description

The current coset FFT is inconsistent between parallel and serial versions, which led to hard to debug bugs. (We suspect due to bit reversal issues). This reverts to a simpler parallel coset compute powers method, add tests for parallel/serial consistency.

We should re-investigate how to apply the faster compute powers method without causing errors.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (main)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md` - N/A since its a bug introduced this update
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by Pratyush** [comment]
This LGTM, but let's file an issue to follow up about the bit-reversal issue

**Review by Pratyush — APPROVED** [review]

**Comment by Pratyush** [comment]
Ah one sec; doesn't this PR also affect the radix2 FFT?

**Comment by ValarDragon** [comment]
Changing compute powers does affect the subgroup radix 2 FFT

File: poly/src/domain/radix2/fft.rs [hunk 1]
```diff
@@ -61,12 +62,68 @@ impl<F: FftField> Radix2EvaluationDomain<F> {
...
     #[cfg(not(feature = "parallel"))]
     fn roots_of_unity(&self, root: F) -> Vec<F> {
-        crate::domain::utils::compute_powers_serial(self.size as usize, root)
+        compute_powers_serial(self.size as usize, root)
     }
 
     #[cfg(feature = "parallel")]
     fn roots_of_unity(&self, root: F) -> Vec<F> {
-        crate::domain::utils::compute_powers(self.size as usize, root)
+        // TODO: Understand why this functions output isn't domain.elements(),
```
Comment (Pratyush) [inline poly/src/domain/radix2/fft.rs:70]: The corresponding test fails? If so we should add it anyway, but mark it with an `#[ignore]` attribute

Comment (ValarDragon) [reply poly/src/domain/radix2/fft.rs:70]: added!

**Pull request merged**